### PR TITLE
chore(kuma-cp) certificates over ADS

### LINKED ISSUE #719: Secrets over ADS
### Summary

Right now we have separate XDS and SDS servers.

We create a gRPC stream per TLS context. This means that if for example, a Dataplane has 5 inbound listeners and 1000 outbound (transparent proxy case) we create (5 + 1000) * 2 (one stream for CA cert and one for Identity cert) gRPC streams per Dataplane.

The number of streams can quickly pile up with many Dataplanes and services. 
It seems that the only way to reduce number of streams is to serve secrets over ADS (one stream) with XDS configuration.

Additionally, this would simplify the system since there will be one less server to run.

To do such a change, the XDS server has to be always served over TLS (which we want to do anyway) and we need to move authentication from SDS to XDS (which would make Kuma more secure).

### PR DESCRIPTION
**Pull request by jakubdyszkiewicz**
### Summary

This PR changes the way how certificates are sent to the data plane proxy.
So far, Envoy was configured to with SDS in a way that it opens a new gRPC stream for each cluster and inbound listener.
It was this way for historic reasons (SDS on separate port, auth only on SDS). Since that time, many things changed.
This solution configures Envoy to exchange secrets over ADS just like with any other config.

Here are the benefits of this approach
* Everything is served over one gRPC stream. Imagine 1000 Envoys with 1000 mTLS clusters. So far that meant 1M streams opened. With this change, we have 1000 streams.
  Less streams = better performance.
* Reduce the number of generated certificates. Because every Cluster was using a new stream, with many instances of the control plane, Envoy could connect to different instances.
  In this case, every instance of the control plane had to generate an identity certificate! That might have been confusing for users why there are more certs generated than expected number.
* Now Envoy is now truly connected to one instance of control plane. It's easier to reason about CP<->DP connection and exchange.
* It simplifies the code.

### Full changelog

* Drop pkg/sds
* Drop SDS config
* Introduce pkg/xds/secrets package by bringing Providers from pkg/sds
* Drop ConnectionInfoTracker, it's not needed anymore.
* Drop SDSLocation and TLS cert, it's not needed anymore.

### Issues resolved

Fix #719

### Documentation

- [X] No docs, internal changes only

### Leftover

- Adjust Prometheus dashboards. I will do this in a separate PR.

### Testing

- [X] Unit tests
- [X] E2E tests
- [X] Manual testing on Universal
- [ ] Manual testing on Kubernetes 

### Backwards compatibility

- [ ] Technically it is backwards compatible, there is no action from user. Although this is significant change. Should we backport it?

### CODE REVIEW

**Review by lobkovilya — APPROVED** [review]
Awesome! So much simpler without SDS 👍

File: pkg/xds/sync/dataplane_watchdog.go [hunk 1]
```diff
@@ -96,11 +99,18 @@ func (d *DataplaneWatchdog) syncDataplane() error {
 	if err != nil {
 		return err
 	}
-	if d.lastHash != "" && snapshotHash == d.lastHash {
-		// Kuma policies (including Dataplanes and Mesh) has not change therefore there is no need to regenerate configuration.
+	certInfo := d.secrets.Info(d.key)
+	syncForCert := certInfo != nil && certInfo.ExpiringSoon() // check if we need to regenerate config because identity cert is expiring soon.
+	syncForConfig := snapshotHash != d.lastHash               // check if we need to regenerate config because Kuma policies has changed.
```
[the remark this answers is not in the document]
Comment (jakubdyszkiewicz) [reply pkg/xds/sync/dataplane_watchdog.go:104]: can snapshot hash ever be `""`? I don't think so, therefore we don't need `d.lastHash != ""`

[the remark this answers is not in the document]
Comment (lobkovilya) [reply pkg/xds/sync/dataplane_watchdog.go:104]: yes, that's true, I can't remember why `d.lastHash != ""` was added

File: pkg/config/dp-server/config.go [hunk 2]
```diff
@@ -10,7 +10,7 @@ import (
 
 var _ config.Config = &DpServerConfig{}
 
-// Dataplane Server configuration that servers API like Bootstrap/XDS/SDS.
+// Dataplane Server configuration that servers API like Bootstrap/XDS.
```
[the remark this answers is not in the document]
Comment (jakubdyszkiewicz) [reply pkg/config/dp-server/config.go:13]: I'll fix it in some of the following PRs

**Comment by jakubdyszkiewicz** [comment]
We decided to not backport it

**Pull request merged**